util/src/test/java/tech/pegasys/teku/util/async/AsyncRunnerTest.java

```diff
@@ -0,0 +1,70 @@
+package tech.pegasys.teku.util.async;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static tech.pegasys.teku.util.Waiter.waitFor;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import org.junit.jupiter.api.Test;
+
+public class AsyncRunnerTest {
+
+  @Test
+  public void testRecurrentTaskCancel() throws Exception {
+    AsyncRunner runner = DelayedExecutorAsyncRunner.create();
+    AtomicInteger counter = new AtomicInteger();
+    SafeFuture<Void> task =
+        runner.runWithFixedDelay(counter::incrementAndGet, 100, TimeUnit.MILLISECONDS);
+    waitFor(() -> assertThat(counter).hasValueGreaterThan(3));
+    task.cancel(false);
+    Thread.sleep(100); // task may be running during the cancel() call
```

Comment:
A fixed sleep time here is likely to cause us grief in CI.  Is it possible to test this with `StubAsyncRunner` so we can control exactly when future tasks execute?

Reply:
Changed it in a bit another way: https://github.com/PegaSysEng/teku/pull/1729/commits/76158c80682ada690b642fc46926a9fd079776de